great_expectations/core/http.py

```diff
@@ -0,0 +1,60 @@
+import requests
+from requests.adapters import HTTPAdapter, Retry
+
+from great_expectations import __version__
+
+DEFAULT_TIMEOUT = 20
+
+
+class TimeoutHTTPAdapter(HTTPAdapter):
+    # https://stackoverflow.com/a/62044100
+    # Session-wide timeouts are not supported by requests
+    # but are discussed in detail here: https://github.com/psf/requests/issues/3070
+    def __init__(self, *args, **kwargs):
+        if "timeout" in kwargs:
+            self.timeout = kwargs["timeout"]
```

Comment:
added `pop`. Also added a default.

Didn't want to modify this too much since I straight up pulled it from SO but I think these changes are appropriate